docs/source/entities.mdx

```diff
@@ -0,0 +1,316 @@
+      return fetchProductByID(productRepresentation.id);
+    }
+  },
+  // ...other resolvers...
+}
+```
+
+Let's break this example down:
+
+* You declare an entity's reference resolver in your resolver map, as a member of the entity's corresponding object.
+* A reference resolver's name is always `__resolveReference`.
+* A reference resolver's first parameter is a **representation** of the entity being resolved.
+    * An entity representation is an object that contains the entity's `@key` fields, plus its `__typename` field. These values are provided by the gateway.
+* A reference resolver is responsible for returning all of the entity fields that this subgraph defines.
+    * In this example, the hypothetical `fetchProductByID` field fetches a particular `Product`'s field data based on its `id`.
+
+_Every_ subgraph that contributes at least one field to an entity must define a reference resolver for that entity.
+
+> To learn more about `__resolveReference` in Apollo Server, see the [API docs](/api/apollo-subgraph/#__resolvereference).
+
+## Contributing entity fields
+
+Any number of different subgraphs can contribute fields to an entity definition. Below, the Products and Inventory subgraphs contribute different fields to the `Product` entity:
+
+<CodeColumns>
+
+```graphql
+# Products subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  name: String!
+  price: Int
+}
+```
+
+```graphql
+# Inventory subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  inStock: Boolean!
+}
+```
+
+</CodeColumns>
+
+When a subgraph contributes entity fields, no _other_ subgraph knows about those fields—only the gateway does thanks to the composed supergraph schema.
+
+By default, each subgraph must contribute _different_ fields, with the important exception of `@key` fields. Otherwise, a composition error occurs. To override this default, see [Resolving another subgraph's field](./entities-advanced/#resolving-another-subgraphs-field).
+
+As mentioned previously, each subgraph that _does_ contribute fields to an entity must [define a reference resolver](#2-define-a-reference-resolver) for that entity.
+
+## Referencing an entity _without_ contributing fields
+
+Your subgraphs can use an entity as a field's return type _without_ contributing any fields to that entity. This requires much less code than the steps in [Defining an entity](#defining-an-entity).
+
+Take a look at this `Product` entity in the Products subgraph:
+
+```graphql
+# Products subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  name: String!
+  price: Int
+}
+```
+
+Now, let's say we want to create a Reviews subgraph that includes the following `Review` type:
+
+```graphql{3}
+# Reviews subgraph
+type Review {
+  product: Product!
+  score: Int!
+}
+```
+
+This is possible! However, this subgraph schema is currently invalid because it doesn't define the `Product` entity.
+
+To fix this, we can add a **stub** of the `Product` entity to the Reviews schema, like so:
```

Comment:
It's a bit late notice, but we actually changed our mind on how we'd like to handle that case. For that, we added a `resolvable` argument to keys, which allow to "mark" key fields but opt-out of needing to be able to resolve references to that key. The idea is that we'll want to change the `Product` definition in the example below into:
```graphql
type Product @key(fields: "id", resolvable: false) {
  id: ID!
}
```
The pros of that is that now an entity can always have a `@key` in all subgraphs, so people's mental model can just "entity === has a @key" in all cases.

To be clear, the example here still work in practice (though it will raise a hint that `Product` is marked as an entity in some subgraphs and not others, which the `@key(..., resolvable: false)` solves as well) but I think we want to push the new pattern described above.

Reply:
Oh, and to clarify, that `resolvable` argument *is* part of preview.